### PR TITLE
Domains: Prompt to upgrade when trying to set domain as primary

### PR DESCRIPTION
**Pull request by eltongo**
#### Changes proposed in this Pull Request

This PR prompts users to upgrade when trying to set a non primary domain to primary, as part of this test: p99Zz8-10j-p2

Screenshot 2020-03-02 at 8 08 23 PM

#### Testing instructions

For a site which has at least one registered domain:

* Verify that you get something similar to what is shown in the screenshot above when the registered domain is not primary
* Verify that no such message is shown for non-registered domains, or for domains that are primary
* Verify that clicking on Upgrade takes you to the correct page
* Ensure that this won't be shown for sites with a plan, or for users not participating in the test
* Check that everything works as expected when listing all sites under `/domains/manage`
* Make sure nothing gets broken :)

### CODE REVIEW

File: client/my-sites/domains/domain-management/list/item.jsx [hunk 1]
```diff
@@ -83,6 +121,9 @@ class ListItem extends React.PureComponent {
 	}
 
 	handleClick = () => {
+		if ( this.props.shouldUpgradeToMakePrimary ) {
```
Comment (daledupreez) [inline client/my-sites/domains/domain-management/list/item.jsx]: I think this is where we're eating/ignoring the click event when `this.props.enableSelection === true` -- as written, we *never* handle the click event, regardless of whether we're in select mode or not.

File: client/my-sites/domains/domain-management/list/item.jsx [hunk 2]
```diff
@@ -30,21 +32,57 @@ class ListItem extends React.PureComponent {
...
+
 	render() {
+		const { busy, enableSelection, shouldUpgradeToMakePrimary } = this.props;
 		const cardClass = classNames( 'domain-management-list-item', {
-			busy: this.props.busy,
+			busy: busy || ( enableSelection && shouldUpgradeToMakePrimary ),
 		} );
 
 		return (
 			<CompactCard className={ cardClass } onClick={ this.handleClick }>
```
Comment (daledupreez) [inline client/my-sites/domains/domain-management/list/item.jsx]: I dug into the code for `CompactCard`/`Card` a bit, and the pointer on hover for the item is driven by the `is-clickable` class on the card, which is present when `props.onClick` is defined for the card. (See https://github.com/Automattic/wp-calypso/blob/master/packages/components/src/card/index.jsx#L43.)

File: client/my-sites/domains/domain-management/list/item.jsx (hunk 1 above)
Comment (eltongo) [reply client/my-sites/domains/domain-management/list/item.jsx]: Eh... there was a merge conflict changing how this stuff works, so this was an untested quick change before pushing.

File: client/my-sites/domains/domain-management/list/item.jsx (hunk 2 above)
Comment (eltongo) [reply client/my-sites/domains/domain-management/list/item.jsx]: Thanks for tracking this down!

**Comment by eltongo** [comment]
For now seems it won't work correctly in `/domains/manage` due to plans not being loaded when listing all sites.

**Comment by eltongo** [comment]
It should now work as expected when visiting `/domains/manage`

**Pull request merged**